builders/interoperability/xcm/xc-registration/self-serve-asset-registration.md

```diff
@@ -0,0 +1,220 @@
+   ```
+
+`assetID` is now ready to pass to `evmForeignAssets.createForeignAsset`.
+
+### Derive the XC-20 address
+
+Convert `assetID` to hex, left-pad it to 32 hex chars, and prepend eight `F`s as follows:
+
+```
+xc20Address = 0xFFFFFFFF + hex(assetId).padStart(32, '0')
+```
+
+The XC-20 address of xcDOT as an example can be calculated like so: 
+
+=== "Formula"
+
+    ```ts
+    const xc20Address = `0xFFFFFFFF${hex(assetId).padStart(32, "0")}`;
+    ```
+=== "Example"
+
+    ```bash
+    0xFFFFFFFF1FCACBD218EDC0EBA20FC2308C778080
+    ```
+
+
+## Generate the Encoded Call Data {: #generate-the-encoded-call-data }
+
+The snippet below shows how to build the call that needs to be sent to Moonbeam that creates the foreign asset. Save the resulting hex string because you will embed it inside a subsequent XCM `Transact` call dispatched from your sibling parachain.
+
+```typescript
+import "@moonbeam-network/api-augment";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { blake2AsHex } from "@polkadot/util-crypto";
+
+const moonbeam = await ApiPromise.create({ provider: new WsProvider(MOONBEAM_WSS) });
+
+const tx = moonbeam.tx.evmForeignAssets.createForeignAsset(
+  ASSET_ID,
+  assetLocation,
+  DECIMALS,
+  SYMBOL,
+  NAME
+);
+
+// SCALE-encoded call data (includes call index 0x3800)
+const encodedCall = tx.method.toHex();
+console.log("Encoded call data:", encodedCall);
+
+// Optional: 32-byte call hash (blake2_256)
+console.log("Call hash:", blake2AsHex(encodedCall));
+```
+
+### Dispatch the call with XCM Transact {: #dispatch-the--call-with-xcm-transact }
+
+To register your asset, your parachain must send a three-instruction XCM v4 message that looks like the following instruction set:
+
+```text
+1. WithdrawAsset { assets: (Here, <fee>), effects: [] }
+2. BuyExecution  { fees:   (Here, <fee>), weightLimit: Unlimited }
+3. Transact {
+     originKind: SovereignAccount,
+     requireWeightAtMost: <weight>,
+     call: <encodedCall>
+   }
+```
+
+- `WithdrawAsset` debits the fee from your parachain’s sovereign account on Moonbeam; `Here` means the payment is in GLMR / MOVR / DEV that the account already holds.
+- `BuyExecution` purchases enough weight for the call. You can also use `Unlimited`.
+- `Transact` executes the SCALE-encoded `createForeignAsset` call as your sovereign account (`originKind: SovereignAccount`), with `<weight>` matching or below the weight you bought.
+
+Send the message with `xcmPallet.send` (also known as `polkadotXcm.send`) from your parachain’s root or governance origin, targeting Parachain `2004` for Moonbeam (`2023` for Moonriver):
+
+```rust
+xcmPallet.send(
+    dest: { Parachain: 2004 },
+    message: VersionedXcm::V4(<program above>)
+);
+```
+
+Finally, look for the following event emitted successfully on Moonbeam:
+
+```
+EvmForeignAssets.ForeignAssetCreated(assetId, location, creator)
+```
+
+Its presence confirms the XC-20 asset is live.
```

Comment:
This is great, but maybe a bit too detailed? I think just saying that they need to execute the bytes from before wrapped in an `Transact` XCM Instruction that is executed with the sovereign account origin. And I also like that you mentioned to look for the event etc